File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/did/KeyForLocalDidTest.java
package org.hyperledger.indy.sdk.did;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class KeyForLocalDidTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testKeyForLocalDidWorksForMyDid() throws Exception {
		DidResults.CreateAndStoreMyDidResult result = Did.createAndStoreMyDid(wallet, "{}").get();
		String did = result.getDid();
		String key = result.getVerkey();

		String receivedKey = Did.keyForLocalDid(wallet, did).get();
		assertEquals(key, receivedKey);
	}

	@Test
	public void testKeyForLocalDidWorksForTheirDid() throws Exception {
		Did.storeTheirDid(wallet, String.format(IDENTITY_JSON_TEMPLATE, DID_MY1, VERKEY_MY1)).get();

		String receivedKey = Did.keyForLocalDid(wallet, DID_MY1).get();
		assertEquals(VERKEY_MY1, receivedKey);
	}

	@Test
	public void testKeyForDidWorksForNoKey() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		Did.keyForLocalDid(wallet, DID_MY2).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/DidTests/KeyForLocalDidTest.cs
using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.DidTests
{
    [TestClass]
    public class KeyForLocalDidTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestKeyForLocalDidWorksForMyDid()
        {
            var result = await Did.CreateAndStoreMyDidAsync(wallet, "{}");
            var did = result.Did;
            var key = result.VerKey;

            var receivedKey = await Did.KeyForLocalDidAsync(wallet, did);

            Assert.AreEqual(key, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForLocalDidWorksForTheirDid()
        {
            var identityJson = string.Format(IDENTITY_JSON_TEMPLATE, DID_MY1, VERKEY_MY1);
            await Did.StoreTheirDidAsync(wallet, identityJson);

            var receivedKey = await Did.KeyForLocalDidAsync(wallet, DID_MY1);

            Assert.AreEqual(VERKEY_MY1, receivedKey);
        }

        [TestMethod]
        public async Task TestKeyForDidWorksForNoKey()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
               Did.KeyForLocalDidAsync(wallet, DID_MY2)
           );
        }
    }
}
